### DISCUSSION TITLE
Why was ogen started?

### DISCUSSION THREAD

**Discussion by guettli** [ogen-go/ogen#783] [General]
We are looking for a way to generate a Go server from an openapi spec.

We are looking at your project and:

* https://github.com/OpenAPITools/openapi-generator
* https://github.com/deepmap/oapi-codegen

Since the other two projects existed before ogen was started, I am curious: 

Why did you start ogen, instead of using one of the existing projects?

**Comment by ernado** [comment]
Probably https://ogen.dev/blog/ogen-intro can help, there is no 1:1 comparision yet, but we were definitely aware about oapi-codegen when starting the project. 

Mostly because of `oneOf` handling, static typing, router-agnostic, support for string "format"-s like datetime, performance of json encoding and decoding. 

Overall, I (and not only me actually) was not satisfied by existing solutions and decided to create a new one. 

Also, the [OpenAPITools/openapi-generator](https://github.com/OpenAPITools/openapi-generator) is in Java and [deepmap/oapi-codegen](https://github.com/deepmap/oapi-codegen) is not so actively maintained as ogen.
